File: 프로그래머스/lv2/12911. 다음 큰 숫자/다음 큰 숫자.js
function solution(n) {
    let answer = n;
    let nBinary = n.toString(2);
    let countOne = nBinary.split("").filter((num) => num != "0").length;
    while(true) {
        answer++;
        let nextBinary = answer.toString(2);
        let nextOneCount = nextBinary.split("").filter((num) => num != "0").length;
        if(nextOneCount == countOne) {
            break;
        }
    }
    return answer;
}

File: 프로그래머스/lv2/12911. 다음 큰 숫자/다음 큰 숫자.java
import java.util.*;
import java.lang.*;
class Solution {
    public int solution(int n) {
        int nBinary = Integer.bitCount(n);
        while(true) {
            n++;
            int nextBinary = Integer.bitCount(n);
            if(nextBinary == nBinary){
                break;
            }
        }
        return n;
    }
    
}